packages/addons-inject/index.js

```diff
@@ -0,0 +1,319 @@
+    'script[src="https://assets.readthedocs.org/static/javascript/readthedocs-analytics.js"]',
+    'script[src="https://assets.readthedocs.org/static/javascript/readthedocs-doc-embed.js"]',
+    'script[src="https://assets.readthedocs.org/static/core/js/readthedocs-doc-embed.js"]',
+  ];
+  static removalLinks = [
+    'link[href="/_/static/css/readthedocs-doc-embed.css"]',
+    'link[href="https://assets.readthedocs.org/static/css/readthedocs-doc-embed.css"]',
+    // Badge only and proxied stylesheets
+    'link[href="https://assets.readthedocs.org/static/css/badge_only.css"]',
+    'link[href="/_/static/css/badge_only.css"]',
+  ];
+  static removalElements = [
+    // Sphinx-rtd-theme version warning
+    "[role=main] > div:first-child > div:first-child.admonition.warning",
+    // Furo version warning
+    "[role=main] > div:first-child.admonition.warning",
+    // Book version warning
+    "#main-content > div > div > article > div:first-child.admonition.warning",
+    // Flyout
+    "div.rst-versions",
+  ];
+
+  // Additional replacements that we perform
+  static replacements = {
+    searchtools: {
+      pattern: `/* Search initialization removed for Read the Docs */`,
+      replacement: `
+/* Search initialization manipulated by Read the Docs using Cloudflare Workers */
+/* See https://github.com/readthedocs/addons/issues/219 for more information */
+
+function initializeSearch() {
+  Search.init();
+}
+
+if (document.readyState !== "loading") {
+  initializeSearch();
+}
+else {
+  document.addEventListener("DOMContentLoaded", initializeSearch);
+}`,
+    },
+  };
+}
+
+// "readthedocsDataParse" is the "<script>" that calls:
+//
+//   READTHEDOCS_DATA = JSON.parse(document.getElementById('READTHEDOCS_DATA').innerHTML);
+//
+const readthedocsDataParse = "script[id=READTHEDOCS_DATA]:first-of-type";
+const readthedocsData = "script[id=READTHEDOCS_DATA]";
+
+async function onFetch(request, env, context) {
+  // Avoid blank pages on exceptions
+  context.passThroughOnException();
+
+  const response = await fetch(request);
+
+  if (response.body === null) {
+    // TODO This was encountered when Cloudflare already has a request cached
+    // and could respond with a 304 response. It's not clear if this is
+    // necessary or wanted yet.
+    console.debug("Response body was already null, passing through.");
+    return response;
+  }
+
+  try {
+    const transformed = await transformResponse(response);
+    // Wait on the transformed body for errors to evaluate. Timeout errors and
+    // errors thrown during transform aren't actually raised until the body is
+    // read. If there was no return, we should return the original response.
+    if (transformed !== undefined) {
+      return new Response(await transformed.arrayBuffer(), transformed);
+    }
+  } catch (error) {
+    console.error("Discarding error:", error);
+    console.debug("Returning original response to avoid blank page");
```

Comment:
I did. Is there a specific concern here?

Reply:
Nope, just wondering if you were able to hit this try/catch in production to know that everything is working as expected -- since I had issues to handle the exceptions even when the code looked good.